src/lib.rs

```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+extern crate smallbitvec;
```

Comment:
You shouldn't need `extern crate` in rust 2018.

Reply:
I didn't enable Rust 2018 yet because it prints a ton of `variable does not need to be mutable` warnings for the `nodes!` macro:

```
warning: variable does not need to be mutable
   --> src/tests.rs:60:13
    |
60  |           let mut item = Item::new(Guid::from($guid), Kind::$kind);
    |               ----^^^^
    |               |
    |               help: remove this `mut`
...
95  |       let remote_tree = nodes!({
    |  _______________________-
96  | |         ("unfiled_____", Folder[needs_merge = true], {
97  | |             ("folderBBBBBB", Folder[needs_merge = true], {
98  | |                 ("bookmarkDDDD", Bookmark[needs_merge = true]),
...   |
108 | |         })
109 | |     }).into_tree().unwrap();
    | |______- in this macro invocation
```

...When `item` definitely needs to be mutable, because we modify its fields!

Reply:
In some cases it doesn't need to be mutable: e.g. when `[ $( $name:ident = $value:expr ),* ]` matches 0 tokens (also when there's 0 children for the macro's other case).

I suspect avoiding the muts in those case will either be difficult and or verbose. Since it's just for tests I'd say add `#[allow(unused_mut)]` to both muts in the macro and enjoy the 2018 edition